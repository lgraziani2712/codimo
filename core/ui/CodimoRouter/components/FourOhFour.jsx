/**
 * @author Luciano Graziani @lgraziani2712
 * @license {@link http://www.opensource.org/licenses/mit-license.php|MIT License}
 *
 * @flow
 */
import React from 'react';
import styled from 'styled-components';

import gameTextUI from 'core/constants/localize/es/gameTextUI';

import EndGameDifficultyURL from '../images/endGameDifficulty.gif';
import EndGameBackgroundURL from '../images/endGameBackground.jpg';

const blurImage = 'data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAAQAAAAECAYAAACp8Z5+' +
                  'AAAAG0lEQVQIW2NkYGD4z8DAwMgABXAGNgGwSgwVAFbmAgXxvZSoAAAAAElFTkSuQmCC';
const Modal = styled.div`
  background: rgb(255, 255, 255);
  display: block;
  min-height: 316px;
  padding: 0 20px 20px;
  width: 630px;
`;
const ModalContainer = styled.div`
  align-items: center;
  display: flex;
  height: 100%;

  &::before {
    background: url(${EndGameBackgroundURL});
    background-size: cover;
    content: '';
    height: 100%;
    position: absolute;
    top: 0;
    width: 100%;
    z-index: -1;
  }

  &::after {
    background: url(${blurImage}) repeat;
    content: '';
    height: 100%;
    position: absolute;
    top: 0;
    width: 100%;
    z-index: -1;
  }
`;
const ModalImage = styled.img`
  display: block;
`;

type FourOhFour$Props = {|
  location: Object,
|};
/**
 * The temporal End Game announcer.
 *
 * @version 1.0.0
 * @param {Object} location React-router location object.
 * @class
 */
export default function FourOhFour({ location }: FourOhFour$Props) {
  const actualDifficulty = location.pathname.split('/')[2];

  return (
    <ModalContainer className="swal2-fade swal2-shown">
      <Modal role="dialog" className="swal2-modal swal2-show">
        <ModalImage className="swal2-image" src={EndGameDifficultyURL} />
        <h2 className="swal2-title">{gameTextUI.endGameMessage.title}</h2>
        <div className="swal2-content">
          {gameTextUI.endGameMessage.text(actualDifficulty)}
        </div>
      </Modal>
    </ModalContainer>
  );
}
