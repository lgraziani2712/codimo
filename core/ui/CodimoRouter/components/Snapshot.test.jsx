/**
 * @author Luciano Graziani @lgraziani2712
 * @license {@link http://www.opensource.org/licenses/mit-license.php|MIT License}
 *
 * @flow
 * @jest-environment node
 */
import React from 'react';
import renderer from 'react-test-renderer';

import 'test/BlocklyMock';
import 'jest-styled-components';

import FourOhFour from './FourOhFour';

describe('CodimoRouter components snapshots', () => {
  it('renders the raw components correctly', () => {
    const fourOhFourTree = renderer.create(
      <FourOhFour
        location={{ pathname: 'testActivity/testLevel/001' }}
      />,
    ).toJSON();

    expect(fourOhFourTree).toMatchSnapshot();
  });
});
