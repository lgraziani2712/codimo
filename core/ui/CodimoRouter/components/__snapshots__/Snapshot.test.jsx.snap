// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CodimoRouter components snapshots renders the raw components correctly 1`] = `
.c1 {
  background: rgb(255,255,255);
  display: block;
  min-height: 316px;
  padding: 0 20px 20px;
  width: 630px;
}

.c0 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  height: 100%;
}

.c0::before {
  background: url(endGameBackground.jpg);
  background-size: cover;
  content: '';
  height: 100%;
  position: absolute;
  top: 0;
  width: 100%;
  z-index: -1;
}

.c0::after {
  background: url(data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAAQAAAAECAYAAACp8Z5+AAAAG0lEQVQIW2NkYGD4z8DAwMgABXAGNgGwSgwVAFbmAgXxvZSoAAAAAElFTkSuQmCC) repeat;
  content: '';
  height: 100%;
  position: absolute;
  top: 0;
  width: 100%;
  z-index: -1;
}

.c2 {
  display: block;
}

<div
  className="swal2-fade swal2-shown c0"
>
  .c0 {
  background: rgb(255,255,255);
  display: block;
  min-height: 316px;
  padding: 0 20px 20px;
  width: 630px;
}

.c1 {
  display: block;
}

<div
    className="swal2-modal swal2-show c0"
    role="dialog"
  >
    .c0 {
  display: block;
}

<img
      className="swal2-image c0"
      src="endGameDifficulty.gif"
    />
    <h2
      className="swal2-title"
    >
      ¡GENIAL!
    </h2>
    <div
      className="swal2-content"
    >
      ¡TERMINASTE TODOS LOS EJERCICIOS DEL undefined!
    </div>
  </div>
</div>
`;
